### ISSUE TITLE
When filtered, some data is not drawn, yet it's filtered

### ISSUE THREAD

**Issue by LuisBreMa** [okviz/SmartFilter#7]
Greetings.

While I was trying the Smart Filter, I noticed that it behaves differently if you are using the PowerBi Desktop or the published version.

I have several series of data grouped by some given tag (in this case, **curva**), it should look like this when a filter is applied:
![image](https://user-images.githubusercontent.com/40607259/41925802-df5acc34-792a-11e8-9810-1e2f614fa799.png)

However, when I publish and try to filter the data on the web page, most of the series get "lost" (I can only see one plotted line, when there should be 7) from the graph, but the info remains (it seems) because when I hover over the remaining line, I can see the values of the other lines:

![image](https://user-images.githubusercontent.com/40607259/41926131-b9691462-792b-11e8-9c26-34a722d82449.png)

Oddly, if I publish the report with the filter applied, it works ok the first time you access it, but as soon as you interact with it (changing the filter), it goes the wrong way again.

Thank you very much in advance!

Regards,

LB.

**Comment by danieleperilli** [comment]
Can you share your report? Please, send it at support@okviz.com

**Comment by LuisBreMa** [comment]
I will try to build a new project with  similar specs, since this one uses DirectQuery. As soon as I have it I will send it to you.

Thank you!

Regards,

LB.

**Comment by danieleperilli** [comment]
I received your report, but I'm not able to reproduce the issue. The visual works properly in the report inside Power BI online and when published on the web.

**Comment by LuisBreMa** [comment]
Hi again!

I ran a few testing cases last night, the problem is not with your component but with the Power BI Line Chart itself. If you are using Categorical Type in the X axis, and you have two tags for your data and one of this tags, when filtered, forces the graph to use the horizontal scroll, but the other doesn't (because it has fewer values), then you get the problem: the set of data that does not need the horizontal scroll will be drawn in funny incorrect ways if you toggle the filter several times. If both tags force the horizontal scroll, it works ok, I think.
I already opened a ticket to fix this within the PowerBI service (Power BI Desktop will work perfectly, tough).

Thank you very much!

Regards,

LB.
